CI failures on Windows: --only-deps, --no-deps and others

### Checklist

- [X] I added a descriptive title
- [X] I searched for other issues and couldn't find a solution or duplication
- [X] I already searched in Google and didn't find any good information or help
- [X] I looked at the [docs](https://docs.conda.io/projects/conda/en/latest/user-guide/index.html) and didn't see anything to help

### What happened?

Looks like something changed (and most likely _not_ in `conda` itself): three tests are now failing on Windows, regardless the Python version:

```
FAILED tests/test_create.py::IntegrationTests::test_install_only_deps_flag
FAILED tests/test_create.py::IntegrationTests::test_install_update_deps_flag
FAILED tests/test_create.py::IntegrationTests::test_install_update_deps_only_deps_flags
```

I looked into it thinking it was my PR fault, but it's also happening in `master` since Jan 19, 2022. Check CI runs for 4f9c8c01c2af0adcde40d6bef6ff40347a0bf645.

The tests expect Python 3.6 to be installed in that fresh environment but instead 2.7 gets pulled in. It might be a change in the repodata? Did we patch `defaults` recently?

### Conda info

```shell
Check CI run: https://github.com/conda/conda/runs/4872802034?check_suite_focus=true
```


### Conda config

```shell
Check CI run: https://github.com/conda/conda/runs/4872802034?check_suite_focus=true
```


### Conda list

```shell
Check CI run: https://github.com/conda/conda/runs/4872802034?check_suite_focus=true
```


### Additional Context

This is not solver specific.

--- a/tests/test_create.py
+++ b/tests/test_create.py
@@ -1256,40 +1256,40 @@ dependencies:
             assert not package_is_installed(prefix, 'flask')
 
     def test_install_update_deps_flag(self):
-        with make_temp_env("flask=0.12", "jinja2=2.9") as prefix:
-            assert package_is_installed(prefix, "python=3.6")
-            assert package_is_installed(prefix, "flask=0.12")
-            assert package_is_installed(prefix, "jinja2=2.9")
+        with make_temp_env("flask=2.0.1", "jinja2=3.0.1") as prefix:
+            assert package_is_installed(prefix, "python=3.9")
+            assert package_is_installed(prefix, "flask=2.0.1")
+            assert package_is_installed(prefix, "jinja2=3.0.1")
 
             run_command(Commands.INSTALL, prefix, "flask", "--update-deps")
-            assert package_is_installed(prefix, "python=3.6")
-            assert package_is_installed(prefix, "flask>0.12")
-            assert package_is_installed(prefix, "jinja2>2.9")
+            assert package_is_installed(prefix, "python=3.9")
+            assert package_is_installed(prefix, "flask>2.0.1")
+            assert package_is_installed(prefix, "jinja2>3.0.1")
 
     def test_install_only_deps_flag(self):
-        with make_temp_env("flask=0.12.2", "jinja2=2.9") as prefix:
-            assert package_is_installed(prefix, "python=3.6")
-            assert package_is_installed(prefix, "flask=0.12.2")
-            assert package_is_installed(prefix, "jinja2=2.9")
+        with make_temp_env("flask=2.0.2", "jinja2=3.0.2") as prefix:
+            assert package_is_installed(prefix, "python=3.9")
+            assert package_is_installed(prefix, "flask=2.0.2")
+            assert package_is_installed(prefix, "jinja2=3.0.2")
 
             run_command(Commands.INSTALL, prefix, "flask", "--only-deps")
-            assert package_is_installed(prefix, "python=3.6")
-            assert package_is_installed(prefix, "flask=0.12.2")
-            assert package_is_installed(prefix, "jinja2=2.9")
+            assert package_is_installed(prefix, "python=3.9")
+            assert package_is_installed(prefix, "flask=2.0.2")
+            assert package_is_installed(prefix, "jinja2=3.0.2")
 
-        with make_temp_env("flask==0.12.2", "--only-deps") as prefix:
+        with make_temp_env("flask==2.0.2", "--only-deps") as prefix:
             assert not package_is_installed(prefix, "flask")
 
     def test_install_update_deps_only_deps_flags(self):
-        with make_temp_env("flask=0.12.2", "jinja2=2.9") as prefix:
-            assert package_is_installed(prefix, "python=3.6")
-            assert package_is_installed(prefix, "flask=0.12.2")
-            assert package_is_installed(prefix, "jinja2=2.9")
+        with make_temp_env("flask=2.0.1", "jinja2=3.0.1") as prefix:
+            assert package_is_installed(prefix, "python=3.9")
+            assert package_is_installed(prefix, "flask=2.0.1")
+            assert package_is_installed(prefix, "jinja2=3.0.1")
 
-            run_command(Commands.INSTALL, prefix, "flask", "python=3.6", "--update-deps", "--only-deps")
-            assert package_is_installed(prefix, "python=3.6")
-            assert package_is_installed(prefix, "flask=0.12.2")
-            assert package_is_installed(prefix, "jinja2>2.9")
+            run_command(Commands.INSTALL, prefix, "flask", "python=3.9", "--update-deps", "--only-deps")
+            assert package_is_installed(prefix, "python=3.9")
+            assert package_is_installed(prefix, "flask=2.0.1")
+            assert package_is_installed(prefix, "jinja2>3.0.1")
 
 
     @pytest.mark.xfail(on_win, reason="nomkl not present on windows",

